Library/OcLogAggregatorLib/OcLog.c

```diff
@@ -363,24 +366,45 @@ InternalLogAddEntry (
     // Write to internal buffer.
     //
 
+    STATIC_ASSERT (OC_LOG_BUFFER_SIZE <= MAX_UINT32, "log buffer size must fit within 32 bits");
+    OldOffset = (UINT32)AsciiStrLen (Private->AsciiBuffer);
```

Comment:
Any clue how long this takes accumulated? Would it be beneficial to record this to `Private` instead of re-determining it?

Reply:
Definitely < 10% of the time we're saving by making this change overall. But yes, can put to private and see.

Reply:
This made no measurable difference at all (change from 3-4ms per log line to 3-4ms per log line... as opposed to 20-40ms with safe write), but I've moved it to private in upcoming revision anyway.